CICD 3.1.0 Changes

cs-abhishek-shukla opened the pull request:
- Changed structure of the doc
- Laid information in an easy to read format
- Added information pertaining to CRs and the flow
- Commit for simultaneous verification

cs-avit-suryawanshi commented on docs/contents.md:
```diff
@@ -119,3 +119,8 @@ The **Continuous Delivery** solution pack installs the following resources:
 - Change Management List
```
In Widgets Section , add below widgets too
1. Playbook Execution Wizard
2. Playbook Buttons
3. Export Template Wizard
4. CICD Content Import

cs-abhishek-shukla replied:
Need descriptions too... @cs-avit-suryawanshi

cs-avit-suryawanshi replied:
Descriptions:
1. Playbook Execution Wizard
    - The Playbook Execution Wizard widget renders playbook execution logs and related comments on the wizard
2. Playbook Buttons
   -  The Playbook Buttons widget creates playbook buttons on a records detailed view
3. Export Template Wizard
   - Manage FortiSOAR export templates that control what content is saved to source control systems like GitHub.
4. CICD Content Import
   - Select a source control repository, review its contents, and import updates into FortiSOAR with full control over applied changes using editable export templates.

cs-abhishek-shukla replied:
Added

cs-sujeet-ahirrao commented on playbooks/02 - Use Case - CICD/Review and Apply Latest Content.json:
```diff
@@ -6,7 +6,7 @@
     "description": "Review and apply latest content from Source Control repository and apply it within the FortiSOAR environment",
     "isActive": true,
-    "debug": false,
+    "debug": true,
```
No need to update this playbooks. Please remove this

cs-sujeet-ahirrao commented on playbooks/02 - Use Case - CICD/Setup CICD Environment.json:
```diff
@@ -1,4 +1,5 @@
 {
+    "@context": "\/api\/3\/contexts\/Workflow",
```
No need to update this playbooks. Please remove this

cs-sujeet-ahirrao commented on postInstall/execution_list.json:
```diff
@@ -0,0 +1,3 @@
+{
```
No need to update this. Please remove this

cs-sujeet-ahirrao commented on postInstall/workflow/CICD Upgrade Script.json:
```diff
@@ -0,0 +1,979 @@
+{
```
No need to update this playbooks. Please remove this

cs-sujeet-ahirrao commented on postInstall/workflow/collection.metadata.json:
```diff
@@ -0,0 +1,11 @@
+{
```
No need to update this. Please remove this

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
 
-    - Map Source Control Username
+    - A **Reset Task** button to reset a stuck task (anything not in *Pending*).
+    - An **Abort** button to stop a task in progress.
```
This button only used when the Change Request is creating and user want to abort it

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
-    - Map Source Control Username
+    - A **Reset Task** button to reset a stuck task (anything not in *Pending*).
+    - An **Abort** button to stop a task in progress.
+  - Analysts gain more flexibility in managing workflows without escalating to administrators.
```
Can you please elaborate this. Why we added this point

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
 
-- **JSON to Grid** widget now has following capabilities:
+  * A new card under both *Production* and *Development* tabs consolidates management of exported content such as playbooks, module definitions, connectors, and related components.
+  * This streamlines updates and ensures critical components are version-controlled.
```
There are two settings are there
1. Export Template which you already explained 
2. Configuration Parameters which we have in cicd_config global variable and now we moved this under settings to mange explicitly if required

cs-abhishek-shukla replied:
Done

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
-    - Show grid columns even if there is no change request record present  
+  * A new button in both *Production* and *Development* environments triggers an **import wizard** that guides analysts through applying content changes.
+  * The wizard applies both *Production Content* and *Production Settings* safely, with a clear preview of what will change.
+  * A new **CR Status** column under issues shows the lifecycle state of each CR, improving at-a-glance tracking.
```
This is the part of BugFixes/Enhancements

cs-abhishek-shukla replied:
Added under miscellaneous

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
 
-- **CICD Configuration Wizard** now offers **GitLab**, along with GitHub, as a source control platform and helps configure the selected source control with *Continuous Delivery* solution pack.
+  - When submitting a PR for review, analysts can now **select reviewers from a drop-down list of Git users** rather than typing usernames manually.
+  - Repositories and PR creation notifications now include **direct links** to the created repository or PR, reducing navigation overhead.
```
Can you please elaborate this.

cs-abhishek-shukla replied:
Added in miscellaneous

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
-## Other Enhancements
+  - Multiple production environments were not handled correctly.
+  - Repository creation could fail if the repository already existed.
+  - Production/Development tabs appeared prematurely before source control setup.
```
- Production/Development tabs will be appeared after the source control setup.

cs-abhishek-shukla replied:
This is issue is fixed and listed under bug "fixes" and hence the wording

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
+  - Repository creation could fail if the repository already existed.
+  - Production/Development tabs appeared prematurely before source control setup.
+  - PR links were not visible in the UI when multiple issues were listed.
+  - Change Management tab visibility was inconsistent.
```
we already covered this .. no need to add

cs-abhishek-shukla replied:
Removed

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
+- Fixed issues where:
 
-## Other Enhancements
+  - Multiple production environments were not handled correctly.
```
Not able to create the separate prod-settings repository while using the multiple production environment using "Source Control Setup" Playbook

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
+  - Multiple production environments were not handled correctly.
+  - Repository creation could fail if the repository already existed.
+  - Production/Development tabs appeared prematurely before source control setup.
+  - PR links were not visible in the UI when multiple issues were listed.
```
PR links and PR Status were not visible in the UI when multiple issues were listed.

cs-abhishek-shukla replied:
Corrected

cs-sujeet-ahirrao commented on release_notes.md:
```diff
@@ -3,56 +3,71 @@
+  - Production/Development tabs appeared prematurely before source control setup.
+  - PR links were not visible in the UI when multiple issues were listed.
+  - Change Management tab visibility was inconsistent.
+  - Certain playbooks failed (e.g., create issue, submit for review).
```
Error handling/Pre checks was improved in playbooks (CICD Taks) such as (e.g., create issue, submit for review).

cs-abhishek-shukla replied:
Added

cs-sujeet-ahirrao commented on README.md:
```diff
@@ -19,99 +19,17 @@ The following diagram simplifies how FortiSOAR's Continuous Delivery works:
 
```
please add like Dev Branch would be the CR-<CR-Number> which will be fork from the Prod branch i.e. main/develop branch

cs-abhishek-shukla replied:
Hidden. Needs rework on the diagram. Parked for later.

cs-sujeet-ahirrao commented on README.md:
```diff
@@ -19,99 +19,17 @@ The following diagram simplifies how FortiSOAR's Continuous Delivery works:
-## Intended Audience
+## Who Should Read This
 
-This document is meant for content developers and FortiSOAR administrators who build FortiSOAR content to suit their daily requirements. Customizations to FortiSOAR content can be pushed to source control in the development environment. After ironing out any issues with customizations and when confident about the custom content, the same can be pushed to the production source control to be delive ...
```
Customizations to FortiSOAR content can be pushed to source control in the development environment. => Customizations to FortiSOAR content can be pushed to the development source control branch.

cs-abhishek-shukla replied:
Content removed

cs-sujeet-ahirrao commented on docs/best-practices.md:
```diff
@@ -0,0 +1,63 @@
+
+    - Your preferred source control platform's username specified during configuration must be mapped with the user who is setting up the Source Control.
+
+2. For repos with restricted merge access, during the initial commit the current logged in user must have sufficient privileges to push the code without raising a PR.
```
Can you please simplified this sentence more

cs-abhishek-shukla replied:
> For repos with restricted merge access, the logged-in user must have sufficient privileges to make the initial commit without creating a PR.

cs-sujeet-ahirrao commented on docs/best-practices.md:
```diff
@@ -0,0 +1,63 @@
+
+9. Following global variables must never be modified or deleted:
+
+    - `cicd_config`
```
Please remove cicd_config global variable as we moved this in Source Control Settings under configuration parameters field and user can modify if he want to change the branch etc.

cs-abhishek-shukla replied:
> 9. The global variable `cicd_env` must never be modified or deleted.

cs-sujeet-ahirrao commented on docs/best-practices.md:
```diff
@@ -0,0 +1,63 @@
+
+14. Changes related to a single change request (CR) must never be pushed from multiple development environments.
+
+15. After the action **Apply Latest Content** is run on development or production environment an **fsrimport-xxx**  file(For example `fsrimport-381e5a85-ccb5-4407-b5ea-97d294bb3a5e`) is created in the `/tmp` directory.
```
No need to add this

cs-abhishek-shukla replied:
Removed

cs-sujeet-ahirrao commented on docs/best-practices.md:
```diff
@@ -0,0 +1,63 @@
+
+15. After the action **Apply Latest Content** is run on development or production environment an **fsrimport-xxx**  file(For example `fsrimport-381e5a85-ccb5-4407-b5ea-97d294bb3a5e`) is created in the `/tmp` directory.
+
+    These files are intended for deletion. To automate the process, you can establish a cron job to remove these files.
```
Remove this point also

cs-abhishek-shukla replied:
removed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+This task entails following sub-tasks:
+
+1. Application administrator creates a new CR for building a new module. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
```
Application administrator creates a new CR in Production Environment for building a new module.

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+This task entails following sub-tasks:
+
+1. Application administrator creates a new CR for building a new module. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
```
Link is not working or md file is missing

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+1. Application administrator creates a new CR for building a new module. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
+
+2. Content Developer creates a new module. For creating a new module, refer to the article [Creating a New Module](https://docs.fortinet.com/document/fortisoar/7.3.1/administration-guide/97786/application-editor#Creating_a_New_Module).
```
Content Developer creates a new module in development environment.

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+    2. For editing permissions assigned to roles, refer to the article [Configuring Roles](https://docs.fortinet.com/document/fortisoar/7.3.1/administration-guide/202940/security-management#Configuring_Roles).
+
+5. Content developer edits the **Source Control &ndash; Production Content** template to include module, roles, picklists, and any other content created to support the new module.
```
Content developer edits the **Source Control &ndash; Production Content** template by going to "Source Control Settings" card and include module, roles, picklists, and any other content created to support the new module.

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+5. Content developer edits the **Source Control &ndash; Production Content** template to include module, roles, picklists, and any other content created to support the new module.
+
+6. Content developer initiates [action on the new Change Request](#working-on-the-new-change-request).
```
link is not working

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+6. Content developer initiates [action on the new Change Request](#working-on-the-new-change-request).
+
+7. Application administrator [Approves & marks as complete the CR in Production environment](#approve--mark-as-complete-a-cr-in-production-environment).
```
link is not working

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+
+7. Application administrator [Approves & marks as complete the CR in Production environment](#approve--mark-as-complete-a-cr-in-production-environment).
+
+8. Application administrator merges latest changes from the source control platform by [Applying Latest changes in Production environment](#apply-latest-changes-in-production-environment).
```
link is not working

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-module-cr.md:
```diff
@@ -0,0 +1,34 @@
+8. Application administrator merges latest changes from the source control platform by [Applying Latest changes in Production environment](#apply-latest-changes-in-production-environment).
+
+> [!NOTE]
+> Once a module is created and applied to production and development, [applying latest changes in production](#apply-latest-changes-in-production-environment) or in [development](#apply-latest-changes-in-development-environment) will not remove it &mdash; even if the module no longer exists in the pulled changes.
```
links are not working

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+
+This task entails following sub-tasks:
+
+1. Application administrator creates a new CR for building a new playbook. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
```
Application administrator creates a new CR in Production Env

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+
+This task entails following sub-tasks:
+
+1. Application administrator creates a new CR for building a new playbook. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
```
link is not working

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+
+1. Application administrator creates a new CR for building a new playbook. For creating a new CR refer to the [New Change Request from Production Environment](#new-change-request-from-production-environment).
+
+2. Content Developer creates a new playbook. For creating new playbooks, refer to [Introduction to playbooks](https://docs.fortinet.com/document/fortisoar/7.3.1/playbooks-guide/331279/introduction-to-playbooks). 
```
Content Developer creates a new playbook in dev env

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+2. Content Developer creates a new playbook. For creating new playbooks, refer to [Introduction to playbooks](https://docs.fortinet.com/document/fortisoar/7.3.1/playbooks-guide/331279/introduction-to-playbooks). 
+
+> [!NOTE]
+> Following actions must be carried out if the playbook requires a new playbook collection:
```
Following actions must be carried out if content developer wants to pushed the a new playbook collection:

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+> [!NOTE]
+> Following actions must be carried out if the playbook requires a new playbook collection:
+> 1. Content developer creates a new collection and underlying playbooks.
+> 2. Content developer edits the export template and adds the newly created playbook collection in **Source Control &ndash; Production Content** export template. For details on editing an export template, refer to [Export Wizard](https://docs.fortinet.com/document/fortisoar/7.3.1/administration-guide/97786/application-editor#Export_Wizard).
```
Source Control – Production Content export template using the Source Control settings

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+> Following actions must be carried out if the playbook requires a new playbook collection:
+> 1. Content developer creates a new collection and underlying playbooks.
+> 2. Content developer edits the export template and adds the newly created playbook collection in **Source Control &ndash; Production Content** export template. For details on editing an export template, refer to [Export Wizard](https://docs.fortinet.com/document/fortisoar/7.3.1/administration-guide/97786/application-editor#Export_Wizard).
+
```
numbering is not in sequence

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+
+4. Content developer initiates [action on the new Change Request](#working-on-the-new-change-request).
+
+5. Application administrator [Approves & marks as complete the CR in Production environment](#approve--mark-as-complete-a-cr-in-production-environment).
```
Link is not working

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+> 1. Content developer creates a new collection and underlying playbooks.
+> 2. Content developer edits the export template and adds the newly created playbook collection in **Source Control &ndash; Production Content** export template. For details on editing an export template, refer to [Export Wizard](https://docs.fortinet.com/document/fortisoar/7.3.1/administration-guide/97786/application-editor#Export_Wizard).
+
+4. Content developer initiates [action on the new Change Request](#working-on-the-new-change-request).
```
Link is not working

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+
+5. Application administrator [Approves & marks as complete the CR in Production environment](#approve--mark-as-complete-a-cr-in-production-environment).
+
+6. Application administrator merges latest changes from the source control platform by [Applying Latest changes in Production environment](#apply-latest-changes-in-production-environment).
```
Link is not working

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/build-playbook-cr.md:
```diff
@@ -0,0 +1,33 @@
+5. Application administrator [Approves & marks as complete the CR in Production environment](#approve--mark-as-complete-a-cr-in-production-environment).
+
+6. Application administrator merges latest changes from the source control platform by [Applying Latest changes in Production environment](#apply-latest-changes-in-production-environment).
+
```
Add the below points
1. Validate the Applied latest contents are working
2. Close the Change Request

cs-abhishek-shukla replied:
Added the first point. Regarding point 2, application admin closed the CR when it was approved and marked complete

cs-sujeet-ahirrao commented on docs/connector-inst-config-git.md:
```diff
@@ -0,0 +1,30 @@
+Connectors and their configuration information can be exported to source control to save time lost in configuration and to keep the sensitive data protected while allowing functionality.
+
+> [!NOTE]
+> For this task the connector must be installed and configured on a production instance.
```
For this task the connector must be installed and configured on a production or development instance.

cs-abhishek-shukla replied:
We should not export connectors configured for a dev environment.

cs-abhishek-shukla replied:
Added info

cs-sujeet-ahirrao commented on docs/connector-inst-config-git.md:
```diff
@@ -0,0 +1,30 @@
+> [!NOTE]
+> For this task the connector must be installed and configured on a production instance.
+
+1. Edit **Source Control &ndash; Production Settings** export template to include connectors to be exported.
```
- Edit **Source Control &ndash; Production Settings** export template using the "Source Control Settings" to include connector which configuration and installation to be exported.

cs-sujeet-ahirrao replied:
We can do same thing for Development instance and saved it in Development Settings

cs-abhishek-shukla replied:
Done

cs-sujeet-ahirrao commented on docs/connector-inst-config-git.md:
```diff
@@ -0,0 +1,30 @@
+## Including Connector Installation and Configuration in Source Control
+
+Connectors and their configuration information can be exported to source control to save time lost in configuration and to keep the sensitive data protected while allowing functionality.
+
```
Need to add notes that user should need to make the dev/prod setting as a private repo

cs-abhishek-shukla replied:
Done

cs-sujeet-ahirrao commented on docs/connector-inst-config-git.md:
```diff
@@ -0,0 +1,30 @@
+
+6. Click **Save** to save the export settings.
+
+7. [Save Production settings](#save-development-settings) to export connectors' installation and configuration to source control.
```
Link is not working and for prod settings [Save Production settings](#save-development-settings) using dev link

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/connector-inst-config-git.md:
```diff
@@ -0,0 +1,30 @@
+
+7. [Save Production settings](#save-development-settings) to export connectors' installation and configuration to source control.
+
+8. [Apply Latest Content](#apply-latest-changes-in-production-environment) to install connectors and import their configurations in a production environment.
```
link is not working

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/contents.md:
```diff
@@ -34,6 +34,10 @@ The **Continuous Delivery** solution pack installs the following resources:
 | CICD Configuration Wizard | Launches a wizard that walks a user through the process of setting up source control for Continuous Delivery.                                                           |
 | Record CTA Tile           | Displays the details of a module record, in Tile form, when the containing button is clicked; and also shows a Call To Action Button to Launch the Record Details Panel |
 | Record Card               | Displays the module records in a card format.                                                                                                                           |
+|Playbook Execution Wizard||
```
Description is missing for new widgets

cs-abhishek-shukla replied:
Added

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+# Editing Export Template
+
+Both the *Production* and *Development* environments have the option to edit what is being written to the production and dev repositories.
```
dev > development

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+1. Setup Continuous Delivery on FortiSOAR environment. Refer [setting up continuous delivery on FortiSOAR](./setup.md#setup-continuous-delivery-on-fortisoar).
+
+2. Setup *Production* and *Dev* environments. Refer [setting up prod environment](./setup.md#setup-development-environment) and [setting up dev environment](./setup.md#setup-development-environment).
```
*Dev* > Development

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+1. Setup Continuous Delivery on FortiSOAR environment. Refer [setting up continuous delivery on FortiSOAR](./setup.md#setup-continuous-delivery-on-fortisoar).
+
+2. Setup *Production* and *Dev* environments. Refer [setting up prod environment](./setup.md#setup-development-environment) and [setting up dev environment](./setup.md#setup-development-environment).
```
[setting up prod environment] link pointing to the dev

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+2. Setup *Production* and *Dev* environments. Refer [setting up prod environment](./setup.md#setup-development-environment) and [setting up dev environment](./setup.md#setup-development-environment).
+
+3. Open the Production or the *Development* tab under **Continuous Delivery**.
```
Production should be *Production* like Development

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+5. Select a template to edit.
+
+    ![](./res/edit-export-template.png)
```
image is not rendering

cs-abhishek-shukla replied:
Not in the diff, but in the doc, yes...
<img width="1815" height="1455" alt="image" src="https://github.com/user-attachments/assets/a06dfc62-c69c-4756-b7ad-5aa954d46fb2" />

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+6. Click the button **Edit Export Template**.
+
+    ![](./res/edit-export-template-select-modules.png)
```
image is not rendering

cs-abhishek-shukla replied:
Not in the diff, but in the doc, yes...

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+7. Select the modules to export and click **Continue**.
+
+    ![](./res/edit-export-template-select-records.png)
```
image is not rendering

cs-abhishek-shukla replied:
Not in the diff, but in the doc, yes...

cs-sujeet-ahirrao commented on docs/editing-export-template.md:
```diff
@@ -0,0 +1,37 @@
+
+8. Select the modules and views to export and click **Continue**.
+
+    ![](./res/edit-export-template-review.png)
```
image is not rendering

cs-abhishek-shukla replied:
Not in the diff, but in the doc, yes...

cs-sujeet-ahirrao commented on docs/export-alerts-incidents-from-prod.md:
```diff
@@ -0,0 +1,45 @@
+
+## Exporting Sample Alerts/Incidents from Prod Environment
+
+Exporting alerts from production environment can help organizations,
```
Exporting alerts/incidents (from SIEM, EDR, etc.) from production environment can help organizations as a backup or archive — primarily to:
1. Reduce storage in the source system, and
2. Preserve historical data for future reference.

cs-abhishek-shukla replied:
Added. Though I do not agree

cs-sujeet-ahirrao commented on docs/export-alerts-incidents-from-prod.md:
```diff
@@ -0,0 +1,45 @@
+
+Following steps help export alerts or incidents from a FortiSOAR production environment to a development environment:
+
+1. Log in to FortiSOAR's production instance.
```
In these step we should need to use the "Source Control Settings" card to edit the repos

cs-abhishek-shukla replied:
Corrected

cs-sujeet-ahirrao commented on docs/export-alerts-incidents-from-prod.md:
```diff
@@ -0,0 +1,45 @@
+
+    6. Click **Save** to save the export settings.
+
+5. [Save development settings](#save-development-settings) to export alerts to source control.
```
- Link is not working
- From Dev and Prod env we can do this operation

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/export-alerts-incidents-from-prod.md:
```diff
@@ -0,0 +1,45 @@
+
+5. [Save development settings](#save-development-settings) to export alerts to source control.
+
+6. [Apply Latest Content](#apply-latest-changes-in-development-environment) to import alerts in development environment. 
```
- link is not working
- Import can do using the prod or dev env

cs-abhishek-shukla replied:
fixed

cs-sujeet-ahirrao commented on docs/initial-source-control-setup.md:
```diff
@@ -0,0 +1,53 @@
+
+    - [Creating branches in a repository - GitLab](https://docs.gitlab.com/ee/user/project/repository/branches/)
+
+5. **Staging, Committing, and pushing changes**: Refer to these articles on contributing and collaborating:
```
As CICD point of view this point is not useful

cs-abhishek-shukla replied:
extra information, for users who would like to know what's happening.

cs-sujeet-ahirrao commented on docs/pluggable-source-control.md:
The source control connector must have following information within `info.json`:
    - A category `Source Code Management`
    - A configuration parameter `username` >> This point is not required

cs-abhishek-shukla replied:
Removed

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
 ## Prerequisites
 
-The **Continuous Delivery** solution pack depends on the following solution packs that are installed automatically &ndash; if not already installed.
+The **Continuous Delivery** solution pack depends on the following solution packs. FortiSOAR automatically installs these dependencies if missing.
```
The **Continuous Delivery** solution pack depends on the following solution packs >> The **Continuous Delivery** solution pack depends on the following solutions

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
 | GitHub         | Connector     | v1.2.0  | Required for source control through GitHub       |
 | GitLab         | Connector     | v2.0.0  | Required for source control through GitLab       |
-| CICD Utils     | Connector     | v1.1.0  | Required for performing CICD specific operations |
+| CICD Utils     | Connector     | v1.1.0  | Required for performing Continuous Delivery specific operations |
```
Add widgets also or removes the connector list and use solution pack word in above comment

cs-abhishek-shukla replied:
Removed connectors

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
-    </td>
-</table>
+> [!NOTE]
+> 1. Only one source control connector (GitHub or GitLab) should be installed to avoid conflicts.
```
Now GitLab and GitHub is installed by default and hence this statement should be like configured at-least one Source Control

cs-abhishek-shukla replied:
Fixed

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
-</table>
+> [!NOTE]
+> 1. Only one source control connector (GitHub or GitLab) should be installed to avoid conflicts.
+> 2. Development and Production environments in FortiSOAR must have the same license type.
```
also add one more point
3. Development instance should be the clone of production instance like same modules, picklists, playbooks before setup the development instance

cs-abhishek-shukla replied:
Added the following in `setup-dev-env.md`
> [!TIP]
> Once the *Development* environment is set up, click the card **Apply Latest Content* to make the development instance a clone of *Production* instance (Required if the production and development FortiSOAR instances are separate).

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
+> 2. Development and Production environments in FortiSOAR must have the same license type.
 
-Following are some prerequisites for using this solution pack:
+### Organizational account
```
We already cover this in 
docs/initial-source-control-setup.md

cs-abhishek-shukla replied:
It is a prerequisite and hence in `setup.md`

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
+- [Create the organization parent group and subgroups - GitLab](https://docs.gitlab.com/ee/tutorials/manage_user/#create-the-organization-parent-group-and-subgroups)
 
-2. **User Account**: Users must have their respective source control accounts. You can then use their usernames to invite them to your organization.
+### User Account
```
We already cover this in 
docs/initial-source-control-setup.md

cs-abhishek-shukla replied:
It is a prerequisite and hence in `setup.md`

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
+- [Add users to a project - GitLab](https://docs.gitlab.com/ee/user/project/members/#add-users-to-a-project)
 
-    - [Permissions and roles - GitLab](https://docs.gitlab.com/ee/user/permissions.html)
+### Source control platform's user roles
```
We already cover this in 
docs/initial-source-control-setup.md

cs-abhishek-shukla replied:
It is a prerequisite and hence in `setup.md`

cs-sujeet-ahirrao commented on docs/setup.md:
```diff
@@ -13,72 +13,67 @@
 
-    1. **Production Content Repository**: Production content, dev content, staging and test changes reside on the *same* repository. To avoid accidental merges and conflicts, it is recommended that this branch be protected by using pull requests to merge developmental changes.
+> [!NOTE]
+> If the production FortiSOAR instance is separate from the development FortiSOAR instance, the content developers working in development environment must also be added in production FortiSOAR instance, and their respective source control usernames be appropriately mapped in connector configurations.
```
can we highlight these 2 points

cs-abhishek-shukla replied:
Added hyperlinks

cs-abhishek-shukla committed:
CICD 3.1.0 Changes
- Changed structure of the doc
- Laid information in an easy to read format
- Added information pertaining to CRs and the flow
- Release notes
- Added matter to other sections
- Updated Screenshots
- Review comments
- Edited upgrade instructions
